[Merged by Bors] - chore(algebra/ordered_field): cleanup

urkud opened the pull request:
* drop `mul_zero_lt_mul_inv_of_pos` and `mul_zero_lt_mul_inv_of_neg`;
* add `iff` lemmas `one_div_pos/neg/nonneg/nonpos` replacing old
  implications;
* some lemmas now assume `≤` instead of `<`;
* rename `mul_lt_of_gt_div_of_neg` to `mul_lt_of_neg_of_div_lt`
  replacing `>` by `<`.
* add `mul_sub_mul_div_mul_neg_iff` and `mul_sub_mul_div_mul_nonpos_iff`,
  generate implications using `alias`;
* drop `div_pos_of_pos_of_pos` (use `div_pos`) and
  `div_nonneg_of_nonneg_of_pos` (use `div_nonneg`);
* replace `div_nonpos_of_nonpos_of_pos` with
  `div_nonpos_of_nonpos_of_nonneg`;
* replace `div_nonpos_of_nonneg_of_neg` with
  `div_nonpos_of_nonneg_of_nonpos`;
* replace `div_mul_le_div_mul_of_div_le_div_pos` and
  `div_mul_le_div_mul_of_div_le_div_pos'` with
  `div_mul_le_div_mul_of_div_le_div_nonneg`; I failed to find
  in the history why we had two equivalent lemmas.
* merge `le_mul_of_ge_one_right` and `le_mul_of_one_le_right'` into
  `le_mul_of_one_le_right`, rename old `le_mul_of_one_le_right` to
  `le_mul_of_one_le_right'`;
* ditto with `le_mul_of_ge_one_left`, `le_mul_of_one_le_left`, and
  `le_mul_of_one_le_left'`;
* ditto with `lt_mul_of_gt_one_right`, `lt_mul_of_one_lt_right`, and
  `lt_mul_of_one_lt_right'`;
* rename `div_lt_of_mul_lt_of_pos` to `div_lt_of_pos_of_lt_mul`;
* rename `div_lt_of_mul_gt_of_neg` to `div_lt_of_neg_of_mul_lt`;
* rename `mul_le_of_div_le_of_neg` to `mul_le_of_neg_of_div_le`;
* rename `div_le_of_mul_le_of_neg` to `div_le_of_neg_of_mul_le`;
* rename `div_lt_div_of_lt_of_pos` to `div_lt_div_of_pos_of_lt`, swap
  arguments;
* rename `div_le_div_of_le_of_pos` to `div_le_div_of_pos_of_le`, swap
  arguments;
* rename `div_lt_div_of_lt_of_neg` to `div_lt_div_of_neg_of_lt`, swap
  arguments;
* rename `div_le_div_of_le_of_neg` to `div_le_div_of_neg_of_le`, swap
  arguments;
* rename `one_div_lt_one_div_of_lt_of_neg` to
  `one_div_lt_one_div_of_neg_of_lt`;
* rename `one_div_le_one_div_of_le_of_neg` to
  `one_div_le_one_div_of_neg_of_le`;
* rename `le_of_one_div_le_one_div_of_neg` to
  `le_of_neg_of_one_div_le_one_div`;
* rename `lt_of_one_div_lt_one_div_of_neg` to
  `lt_of_neg_of_one_div_lt_one_div`;
* rename `one_div_le_of_one_div_le_of_pos` to
  `one_div_le_of_pos_of_one_div_le`;
* rename `one_div_le_of_one_div_le_of_neg` to
  `one_div_le_of_neg_of_one_div_le`.

---
<!-- put comments you want to keep out of the PR commit here -->

urkud committed:
chore(algebra/ordered_field): cleanup

* drop `mul_zero_lt_mul_inv_of_pos` and `mul_zero_lt_mul_inv_of_neg`;
* add `iff` lemmas `one_div_pos/neg/nonneg/nonpos` replacing old
  implications;
* some lemmas now assume `≤` instead of `<`;
* rename `mul_lt_of_gt_div_of_neg` to `mul_ot_of_div_lt_of_neg`
  replacing `>` by `<`.
* add `mul_sub_mul_div_mul_neg_iff` and `mul_sub_mul_div_mul_nonpos_iff`,
  generate implications using `alias`;
* drop `div_pos_of_pos_of_pos` (use `div_pos`) and
  `div_nonneg_of_nonneg_of_pos` (use `div_nonneg`);
* replace `div_nonpos_of_nonpos_of_pos` with
  `div_nonpos_of_nonpos_of_nonneg`;
* replace `div_nonpos_of_nonneg_of_neg` with
  `div_nonpos_of_nonneg_of_nonpos`;
* replace `div_mul_le_div_mul_of_div_le_div_pos` and
  `div_mul_le_div_mul_of_div_le_div_pos'` with
  `div_mul_le_div_mul_of_div_le_div_nonneg`; I failed to find
  in the history why we had two equivalent lemmas.

sgouezel commented on src/algebra/ordered_field.lean:
```diff
@@ -16,64 +16,90 @@ variable {α : Type*}
-have hbinv : 1 / b > 0, from  one_div_pos_of_pos hb,
+lemma one_le_div_of_le (a : α) {b : α} (hb : 0 < b) (h : b ≤ a) : 1 ≤ a / b :=
+have hb'   : b ≠ 0,     from ne_of_gt hb,
+have hbinv : 0 < 1 / b, from  one_div_pos.2 hb,
```
```suggestion
have hbinv : 0 < 1 / b, from one_div_pos.2 hb,
```

sgouezel commented on src/algebra/ordered_field.lean:
```diff
@@ -104,174 +130,75 @@ div_mul_cancel b (ne_of_lt hc) ▸ mul_le_mul_of_nonpos_right h (le_of_lt hc)
     ... = b / c           : eq.symm $ div_eq_mul_one_div b c
 
-lemma mul_lt_of_gt_div_of_neg (hc : c < 0) (h : a > b / c) : a * c < b :=
+lemma mul_lt_of_div_lt_of_neg (hc : c < 0) (h : b / c < a) : a * c < b :=
```
The name of the lemma does not match the order of its assumptions. I don't know if the best fix is to change the name, or the order of the assumptions

sgouezel commented on src/algebra/ordered_field.lean:
```diff
@@ -104,174 +130,75 @@ div_mul_cancel b (ne_of_lt hc) ▸ mul_le_mul_of_nonpos_right h (le_of_lt hc)
 div_mul_cancel b (ne_of_lt hc) ▸ mul_lt_mul_of_neg_right h hc
 
-lemma div_lt_of_mul_lt_of_pos (hc : c > 0) (h : b < a * c) : b / c < a :=
+lemma div_lt_of_mul_lt_of_pos (hc : 0 < c) (h : b < a * c) : b / c < a :=
```
ditto

sgouezel commented on src/algebra/ordered_field.lean:
```diff
@@ -104,174 +130,75 @@ div_mul_cancel b (ne_of_lt hc) ▸ mul_le_mul_of_nonpos_right h (le_of_lt hc)
+   ... > b * (1 / c)     : mul_lt_mul_of_pos_right h (one_div_pos.2 hc)
    ... = b / c           : eq.symm $ div_eq_mul_one_div b c
 
 lemma div_lt_of_mul_gt_of_neg (hc : c < 0) (h : a * c < b) : b / c < a :=
```
here again the name of the lemma is strange (in particular, `mul_gt` should be `mul_lt`)

sgouezel commented on src/algebra/ordered_field.lean:
```diff
@@ -387,16 +307,16 @@ lt_of_not_ge $ λ hn, not_le_of_gt hl $ one_div_le_one_div_of_le h hn
 lt_of_not_ge $ λ hn, not_le_of_gt hl $ one_div_le_one_div_of_le_of_neg h hn
 
-lemma one_div_le_of_one_div_le_of_pos {a b : α} (ha : a > 0) (h : 1 / a ≤ b) : 1 / b ≤ a :=
+lemma one_div_le_of_one_div_le_of_pos {a b : α} (ha : 0 < a) (h : 1 / a ≤ b) : 1 / b ≤ a :=
```
order again

sgouezel replied:
I think the lemma name is still wrong here.

urkud replied:
I don't understand why I missed this one. Searched for `_of_pos ` and `of_neg `, renamed more lemmas.

urkud committed:
Update src/algebra/ordered_field.lean

Co-authored-by: sgouezel <sebastien.gouezel@univ-rennes1.fr>

urkud committed:
Fix more lemma names

* merge `le_mul_of_ge_one_right` and `le_mul_of_one_le_right'` into
  `le_mul_of_one_le_right`, rename old `le_mul_of_one_le_right` to
  `le_mul_of_one_le_right'`;
* ditto with `le_mul_of_ge_one_left`, `le_mul_of_one_le_left', and
  `le_mul_of_one_le_left'`;
* ditto with `lt_mul_of_gt_one_right`, `lt_mul_of_one_lt_right`, and
  `lt_mul_of_one_lt_right'`;
* rename `mul_lt_of_div_lt_of_neg` to `mul_lt_of_neg_of_div_lt`;
* rename `div_lt_of_mul_lt_of_pos` to `div_lt_of_pos_of_lt_mul`;
* rename `div_lt_of_mul_gt_of_neg` to `div_lt_of_neg_of_mul_lt`;

urkud committed:
Merge branch 'master' into ordered-field

urkud committed:
Fix more names

* rename `mul_le_of_div_le_of_neg` to `mul_le_of_neg_of_div_le`;
* rename `div_le_of_mul_le_of_neg` to `div_le_of_neg_of_mul_le`;
* rename `div_lt_div_of_lt_of_pos` to `div_lt_div_of_pos_of_lt`, swap
  arguments;
* rename `div_le_div_of_le_of_pos` to `div_le_div_of_pos_of_le`, swap
  arguments;
* rename `div_lt_div_of_lt_of_neg` to `div_lt_div_of_neg_of_lt`, swap
  arguments;
* rename `div_le_div_of_le_of_neg` to `div_le_div_of_neg_of_le`, swap
  arguments;
* rename `one_div_lt_one_div_of_lt_of_neg` to
  `one_div_lt_one_div_of_neg_of_lt`;
* rename `one_div_le_one_div_of_le_of_neg` to
  `one_div_le_one_div_of_neg_of_le`;
* rename `le_of_one_div_le_one_div_of_neg` to
  `le_of_neg_of_one_div_le_one_div`;
* rename `lt_of_one_div_lt_one_div_of_neg` to
  `lt_of_neg_of_one_div_lt_one_div`;
* rename `one_div_le_of_one_div_le_of_pos` to
  `one_div_le_of_pos_of_one_div_le`;
* rename `one_div_le_of_one_div_le_of_neg` to
  `one_div_le_of_neg_of_one_div_le`;
